src/reanimated2/layoutReanimation/animationBuilder/Keyframe.ts

```diff
@@ -0,0 +1,199 @@
+import { Easing, EasingFn } from '../../Easing';
+import { withDelay, withSequence, withTiming } from '../../animations';
+import {
+  AnimationFunction,
+  EntryExitAnimationBuild,
+  IEntryExitAnimationBuilder,
+  KeyFrameProps,
+  StyleProps,
+} from './commonTypes';
+
+export interface KeyFramePoint {
+  duration: number;
+  value: number | string;
+  easing?: EasingFn;
+}
+export interface ParsedKeyFramesDefinition {
+  initialValues: StyleProps;
+  keyframes: {
+    [key: string]: KeyFramePoint[];
+  };
+}
+
+export interface KeyFramesDefinition {
+  [key: number]: KeyFrameProps;
+}
+export class Keyframe implements IEntryExitAnimationBuilder {
+  durationV?: number;
+  delayV?: number;
+  definitions: KeyFramesDefinition;
+
+  /*
+    Keyframe definition should be passed in the constructor as the map
+    which keys are between range 0 - 100 (%) and correspond to the point in the animation progress.
+  */
```

Comment:
This information wasn't in docs and it's pretty important IMHO.

Reply:
The comment about the linear easing function is important as well.

Reply:
You're right, I'm gonna add it.